javascripts/handlers/revisions.js

```diff
@@ -0,0 +1,103 @@
+Vue.prototype.window = window
+new Vue({
+  el: '.revisions-container',
+  data: {
+    isEntry: false,
+    revisions: [],
+    selectedRevision: null,
+    viewTypes: {
+      'current': "Aperçu de cette version",
+      'commit_diff': "Modifs apportées par cette version",
+      'diff': "Comparaison avec version actuelle"
+    },
+    displayWikiCode: false,
+    selectedViewType: 'current',
+  },
+  computed: {
+    firstRevision() { return this.revisions[this.revisions.length - 1] },
+    lastRevision() { return this.revisions[0] },
+    restoreUrl() { return `${document.location.search}&restoreRevisionId=${this.selectedRevision.id}` }
+  },
+  mounted() {
+    this.isEntry = this.$el.dataset.isEntry == "1"
+    this.revisions = JSON.parse(this.$el.dataset.revisions).map(rev => {
+      rev.id = parseInt(rev.id)
+      rev.time = new Date(rev.time)
+      rev.timestamp = rev.time.getTime()
+      rev.displayTime = rev.time.toLocaleDateString(window.locale, { 
+        year: '2-digit', month: 'short', day: 'numeric', 
+        hour: 'numeric', minute: 'numeric' 
+      })
+      // initial prop so it gets reactive
+      rev.current_code = ''; rev.current_html = ''; 
+      rev.commit_diff_html = ''; rev.commit_diff_code = '';
+      rev.diff_html = ''; rev.diff_code = '';
+      rev.fullyRetrieved = false
+      return rev
+    })    
+    this.calculateRevisionsPlaceInTimeLine()
+    this.selectedRevision = this.lastRevision
+  },
+  watch: {
+    selectedRevision() {
+      if (this.selectedRevision && !this.selectedRevision.fullyRetrieved) {
+        let url = `?api/pages/${this.selectedRevision.id}&includeDiff=true`
+        $.getJSON(url, (data) => {
```

Comment:
Je ne sais pas si cet appel de JSON sur un wiki avec url courte va fonctionner car l'URI sera du genre /MA-PAGE/
et donc la requête va donner l'uri `/PA-PAGE?api/...` alors qu'il faudrait `/?api`

Y aurait-il moyen de récupérer via javascript la baseurl pour éviter les soucis (genre comme pour aceditor https://github.com/YesWiki/yeswiki/blob/b0a3903324098699b8c35e56b2bd0031cb2c8773/tools/aceditor/actions/__footer.php#L48 que nous pourrions mutualiser pour tout le monde finalement) ?

Reply:
je crois qu'on avait déjà regardé ensemble et que ça marchait non? mais j'ai un doute ! comment je peux tester le short url ?

Reply:
En fait, nous en avions parlé, et ça ne fonctionne pas pour bazarliste dynamic (j'ai testé avec des url courtes) : https://github.com/YesWiki/yeswiki/blob/36690139843e871067c4e4f2ac7f143cc4039a0e/tools/bazar/presentation/javascripts/bazar-list-dynamic.js#L181

Pour y arriver, nous pourrions, 
- déplacer la création de la variable js de aceditor https://github.com/YesWiki/yeswiki/blob/36690139843e871067c4e4f2ac7f143cc4039a0e/tools/aceditor/actions/__footer.php#L48 vers l'action footer globale
- utiliser cette variables js alors disponible dans l'environnement js pour faire `getJSON(baseUrl+'?api...')` si ok pour toi je peux faire un commit dans ce sens sur la présente branche

Reply:
Ok ! tant qu'à faire tu peux peut être plutot créer une variable js `wiki` avec dedans certaines valeurs de $this->config, par exemple `pageTag`, `baseUrl`, `lang` ?

Reply:
fait !